### PR TITLE
Fix SDK text limit (999 bytes), raise tool preview limits, fix hook auth

### PR DESCRIPTION
**Pull request by wmoto-ai**
## Summary
- Confirmed `createStartUpPageContainer` limit is **<1000 UTF-8 bytes** via hardware boundary testing (999 OK, 1000 NG). Adjusted `paginateText` maxBytes from 1000 to 999
- Raised `buildToolPreview()` slice limits from 200/300 to 2000 for Edit/Write/default tool previews (firmware scroll support)
- Added `X-CC-G2-Token` header to PermissionRequest hook for Hub auth, fixing notifications not reaching G2 after Hub restart
- Increased notification list from 10 to 20 items
- Updated tests for new 999-byte boundary

## Test plan
- [x] Hardware boundary test: 日本語333文字(999bytes) → OK, 334文字(1002bytes) → NG
- [x] Hardware boundary test: ASCII 999文字(999bytes) → OK, 1000文字(1000bytes) → NG
- [x] Write 2067文字 → 4チャンク分割でファームウェアスクロール閲覧OK
- [x] Edit長文diff → old/new が2000文字まで表示確認
- [x] PermissionRequest hook auth → Hub再起動後もG2に通知が届くこと確認
- [x] vitest 56テスト全パス

### CODE REVIEW

File: scripts/cc-g2.sh [hunk 1]
```diff
@@ -356,6 +356,28 @@ run_internal_command() {
...
+          *) error "Unknown find-session arg: $1"; exit 1 ;;
+        esac
+      done
+      [ -n "$work_dir" ] || { error "find-session requires --workdir"; exit 1; }
+      local base_name
+      base_name="$(make_tmux_session_name "$work_dir")"
+      # Pick the latest session matching base_name (highest suffix wins)
+      local found=""
+      found=$(tmux list-sessions -F '#{session_name}' 2>/dev/null \
+        | grep "^${base_name}\(-[0-9]*\)\{0,1\}$" | sort -V | tail -1)
```
[the remark this answers is not in the document]
Comment (wmoto-ai) [reply scripts/cc-g2.sh:373]: macOS (Apple sort 2.3-197) supports `-V` (`--version-sort`). Verified working on the target platform.

File: src/glasses-ui.ts [hunk 2]
```diff
@@ -39,21 +39,51 @@ export type NotificationUIState = {
...
+ * SDKの文字数制限はUTF-8バイト基準（日本語3bytes/文字のため
+ * コードポイント数では正確に制御できない）。
+ * SCROLL_TOP/SCROLL_BOTTOM イベントはチャンク境界でのみ発火される。
+ *
+ * コードポイント単位で走査しサロゲートペア安全。
+ * 本文のインデントや空白は保持する（全体のtrimのみ）。
+ */
+const textEncoder = new TextEncoder()
+
+export function paginateText(text: string, maxBytes = 999): string[] {
```
[the remark this answers is not in the document]
Comment (wmoto-ai) [reply src/glasses-ui.ts:55]: Intentionally kept as defensive handling — widening the type would leak null-safety concerns to callers. The `@ts-expect-error` in tests explicitly documents this is an edge case, not normal usage.

File: src/main.ts [hunk 3]
```diff
@@ -872,7 +884,10 @@ async function handleNotifEvent(conn: BridgeConnection, event: EvenHubEvent) {
       } else if (notifState.screen === 'detail') {
         // 詳細画面: スクロールでページ送り＋画面遷移
         if (!notifState.detailItem) return
-        const pageCount = glassesUI.getDetailPageCount(notifState.detailItem.fullText)
+        // ghostリストコンテナからのイベントを無視（detail画面ではtextEventとsysEventのみ有効）
+        if (normalized.source === 'list') return
+        // pageCountはdetail遷移時に算出済み（毎回paginateText()を呼ぶのを避ける）
```
[the remark this answers is not in the document]
Comment (wmoto-ai) [reply src/main.ts]: Fixed: updated comment to accurately describe that detailPages are computed per render, not cached.

File: src/glasses-ui.ts [hunk 4]
```diff
@@ -475,12 +511,14 @@ export function createGlassesUI() {
...
         if (h && b) {
-          log(`G2に通知詳細表示: "${detail.title}" page=${pageIndex + 1}/${pages.length}`)
+          log(`G2に通知詳細表示: "${detail.title}" chunk=${pageIndex + 1}/${pages.length} (${bodyText.length}chars, firmware scroll)`)
           return
         }
         log('G2 textContainerUpgrade に失敗 → ページ再描画へフォールバック')
       }
 
+      // paginateText() のデフォルト maxBytes=999 により、各チャンクは
+      // UTF-8で1000バイト以内（≒文字数も1000以下）に収まっている
```
[the remark this answers is not in the document]
Comment (wmoto-ai) [reply src/glasses-ui.ts]: Fixed: updated comment to say 999 bytes.
